### ISSUE TITLE
Mail with image couldn't be parsed successfully

### ISSUE THREAD

**Issue by markusramsak** [zbateson/mail-mime-parser#95]
The parsing of the attached .eml file (zip compressed so I could upload it) failed ... it seems to have something to do with the "Content-Type: image/png; charset=binary; name="logo.png"

[Test.eml.zip](https://github.com/zbateson/mail-mime-parser/files/3847238/Test.eml.zip)

My source code for parsing the .eml file
```
$rawMessage = file_get_contents($pathToFile . 'Test.eml');
$mailParser = new MailMimeParser();
$message = $mailParser->parse($rawMessage);
$attachments = $message->getAllAttachmentParts();
foreach ($attachments AS $attachment) {
    /**             
      * @var \ZBateson\MailMimeParser\Message\Part\MimePart $attachment
      */
    $content = $attachment->getContent(); // error occurs
    // a lot more unimportant lines of code for what to do with the parsed attachment ...
}
```

**Comment by zbateson** [comment]
Hi @markusramsak 

I was considering the existence of a "charset" attribute on a content-type header to be indicative that the content is _not_ binary.  I've added to that condition so a charset of "binary" is considered binary, but my guess is there may be other instances of this in the wild with other weird choices for charset.

Anyway, I'll release that for you.  Thanks for reporting :)

**Comment by zbateson** [comment]
Released in 1.1.7

**Comment by markusramsak** [comment]
Thank you for your quick reply, but I still get the same error ... :(
`iconv_strlen(): Unknown error (22) [ROOT/vendor/zbateson/mb-wrapper/src/MbWrapper.php, line 397]`

**Comment by zbateson** [comment]
Hi @markusramsak 

Thanks for letting me know -- I've pushed a commit that should resolve this now, sorry about the back-and-forth there was a second part I missed.

I'll wait for you to confirm it first before releasing if you can.

Thanks

**Comment by markusramsak** [comment]
With this change, it seems to work. Would you so kind as to release a new version, so I can use this improved version of your excellent parser. Thanks!!

**Comment by zbateson** [comment]
Excellent, re-released in 1.1.8.  I'll close again, but feel free to comment/reopen if still an issue!
